config/src/config.rs

```diff
@@ -2,38 +2,54 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use crate::{
-    config::ConsensusProposerType::{FixedProposer, MultipleOrderedProposers, RotatingProposer},
     keys::{ConsensusKeyPair, NetworkKeyPairs},
-    seed_peers::{SeedPeersConfig, SeedPeersConfigHelpers},
-    trusted_peers::{
-        ConfigHelpers, ConsensusPeersConfig, ConsensusPrivateKey, NetworkPeersConfig,
-        NetworkPrivateKeys, UpstreamPeersConfig,
-    },
-    utils::{deserialize_whitelist, get_available_port, get_local_ip, serialize_whitelist},
+    seed_peers::SeedPeersConfigHelpers,
+    trusted_peers::{ConfigHelpers, ConsensusPrivateKey, NetworkPrivateKeys},
+    utils::get_available_port,
 };
 use failure::prelude::*;
-use libra_crypto::ValidKey;
 use libra_tools::tempdir::TempPath;
 use libra_types::{
-    transaction::{SignedTransaction, Transaction, SCRIPT_HASH_LENGTH},
+    transaction::{SignedTransaction, Transaction},
     PeerId,
 };
-use parity_multiaddr::Multiaddr;
 use prost::Message;
 use serde::{de::DeserializeOwned, Deserialize, Serialize};
 use std::{
-    collections::HashSet,
     convert::TryFrom,
-    fmt,
     fs::File,
     io::{Read, Write},
     path::{Path, PathBuf},
     str::FromStr,
     string::ToString,
-    time::Duration,
 };
 use toml;
 
+mod admission_control_config;
```

Comment:
wait how does cargo understands that this is not in the same directory, but in `config/`?

Reply:
These are submodules of the config. If they were in the same directory, I would need to have an explicit macro telling rust / cargo where they exist.